Source/Dafny/Compiler-js.cs

```diff
@@ -519,26 +519,31 @@ public JavaScriptCompiler(ErrorReporter reporter)
         }
         DeclareField("Witness", true, true, nt.BaseType, nt.tok, witness.ToString(), w);
       }
+      // In JavaScript, the companion class of a newtype (which is what is being declared here) doubles as a
+      // type descriptor for the newtype. The Default() method for that type descriptor is declared here.
       using (var wDefault = w.NewBlock("static get Default()")) {
         var udt = new UserDefinedType(nt.tok, nt.Name, nt, new List<Type>());
-        var d = TypeInitializationValue(udt, wr, nt.tok, false);
+        var d = TypeInitializationValue(udt, wr, nt.tok, false, false);
         wDefault.WriteLine("return {0};", d);
       }
       return cw;
     }
 
     protected override void DeclareSubsetType(SubsetTypeDecl sst, TargetWriter wr) {
-      var cw = CreateClass(IdProtect(sst.Module.CompileName), IdName(sst), sst, wr) as JavaScriptCompiler.ClassWriter;
+      var cw = (ClassWriter)CreateClass(IdProtect(sst.Module.CompileName), IdName(sst), sst, wr);
       var w = cw.MethodWriter;
+      var udt = UserDefinedType.FromTopLevelDecl(sst.tok, sst);
+      string d;
       if (sst.WitnessKind == SubsetTypeDecl.WKind.Compiled) {
-        var witness = new TargetWriter(w.IndentLevel, true);
-        TrExpr(sst.Witness, witness, false);
-        DeclareField("Witness", true, true, sst.Rhs, sst.tok, witness.ToString(), w);
+        var sw = new TargetWriter(w.IndentLevel, true);
+        TrExpr(sst.Witness, sw, false);
+        DeclareField("Witness", true, true, sst.Rhs, sst.tok, sw.ToString(), w);
+        d = TypeName_UDT(FullTypeName(udt), udt, wr, udt.tok) + ".Witness";
+      } else {
+        d = TypeInitializationValue(udt, wr, sst.tok, false, false);
       }
       using (var wDefault = w.NewBlock("static get Default()")) {
-        var udt = UserDefinedType.FromTopLevelDecl(sst.tok, sst);
-        var d = TypeInitializationValue(udt, wr, sst.tok, false);
-        wDefault.WriteLine("return {0};", d);
+        wDefault.WriteLine($"return {d};");
```

Comment:
Why bother with the using and wDefault temporary here. Just write 
w.NewBlock("static get Default()").WriteLine($"return {d};");

Reply:
Sure